src/System.Private.ServiceModel/tools/setupfiles/EnsureBridgeRunning.cmd

```diff
@@ -30,9 +20,29 @@ if '%BridgeResourceFolder%' == '' (
 
 pushd %~dp0..\..\..\..\bin\wcf\tools\Bridge
 echo Invoking Bridge.exe -require -BridgeResourceFolder:%_bridgeResourceFolder% %*
-call Bridge.exe -require -BridgeResourceFolder:%_bridgeResourceFolder% %*
+call Bridge.exe -require -BridgeResourceFolder:%_bridgeResourceFolder%  %*
+set BridgeReturnCode=%ERRORLEVEL%
+
+echo Bridge.exe returned exit code %BridgeReturnCode%
+
+if '%BridgeReturnCode%' == '2' goto weStartedBridge
+if '%BridgeReturnCode%' == '1' goto bridgeAlreadyRunning
+echo EnsureBridgeRunning: The attempt to start the Bridge failed with exit code %BridgeReturnCode%
+goto done
+
+:weStartedBridge
+  echo EnsureBridgeRunning: The Bridge was started in response to this request
+  set BridgeStartedByScript=true
+  goto done
+  
+:bridgeAlreadyRunning
+  echo EnsureBridgeRunning: The Bridge was already running
+  set BridgeStartedByScript=false
+  goto done
+  
+:done
 popd
 
-
+endlocal & set BridgeStartedByScript=%BridgeStartedByScript%
 exit /b
```

Comment:
Keeping an env variable (BridgeStartedByScript) floating around between 4 scripts (EnsureBridgeRunning.cmd, ReleaseBridge.cmd, build.cmd and StartBridge.cmd) could be a headache for long term maintenance.

Why don't we just do "exit /b %BridgeReturnCode%" here (may need move endlocal below)? In Build.cmd, it will check/record the errorlevel of this script. At the time of release, it will pass either -reset or -stop to the bridge.exe depending on the earlier errorlevel code. We can get rid of ReleaseBridge.cmd completely. Finally, in the StartBridge.cmd, it will simply ignore the errorlevel code from this scrip. In this way, each script has full control of the lifetime of bridge, all logic is self-contained and there is no magic variables.